elevate-project/controllers/project.js

```diff
@@ -1,29 +1,97 @@
-
-// const createUser = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-// 	return await requesters.post(req.baseUrl, selectedConfig.targetRoute.path, req.body)
-// }
-// const updateUser = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-
-// 	//const filteredRequestBody = requestParser.transformUpdateUserBody(req.body)
-// 	return await requesters.patch(req.baseUrl, selectedConfig.targetRoute.path, req.body, req.headers)
-// }
-
-// const entityTypeRead = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-// 	return await requesters.post(req.baseUrl, selectedConfig.targetRoute.path, req.body, {
-// 		'X-auth-token': req.headers['x-auth-token'],
-// 	})
-// }
-
-const userController = {
-	// createUser,
-	// updateUser,
-	// entityTypeRead,
+const common = require('../constants/common')
+/**
+ * Fetch project templates from projects service.
+ * @name fetchProjectTemplates
+ * @param {Object} req
+ * @param {Object} res
+ * @param {Object} responses  
+ * @returns {JSON} - List of project templates
+ */
+/**
+ * req.body = {
+ * 		organization_id : 1,
+ * 		resourceType : ['survey','projects']
+ * }
+ */
+
+const fetchProjectTemplates = async (req, res, responses) => {
+	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
+
+	let response = { result: { data: [], count: 0 } }
+	let proceedToCallProjectService = false
+	let resp = {}
+
+	// fetch the max limit from the env file for the DB Find API
+	const max_limit = process.env.RESOURCE_MAX_FETCH_LIMIT ? parseInt(process.env.RESOURCE_MAX_FETCH_LIMIT, 10) : 1000
+
+	if (req.body) {
+		// check if body has key resourceType else assign []
+		const resourceType = req?.body?.resourceType || [];
+		if (Array.isArray(resourceType) && resourceType.length > 0) {
+			// if resource type have type = projects proceed to call api 
+			proceedToCallProjectService = resourceType.includes(common.RESOURCE_TYPE_PTOJECT);
+		}else if(resourceType.length == 0){
+			// if resource type have type = empty call API because the client is expecting all type of resources 
+			proceedToCallProjectService = true	
+		}
+	}
+
+	if (proceedToCallProjectService && req.headers[common.AUTH_TOKEN_KEY]) {
+		let reqBody = {
+			"query": {
+				"status": common.PROJECT_STATUS_PUBLISHED
+			},
+			"projection": common.PROJECT_PROJECTION_FIELDS,
+			"limit": max_limit
+		}
+		
+		// replace the word bearer if token has it 
+		const x_auth_token = req.headers[common.AUTH_TOKEN_KEY].replace(/^(Bearer|bearer)\s*/, '');
+		// custom header 
+		const header = {
+			'internal-access-token' : req.headers['internal_access_token'],
+			'X-auth-token': x_auth_token,
+			'Content-Type': 'application/json'
```

Comment:
@priyanka-TL as you define the content type in the common you can use the same thing here as well
HEADER_CONTENT_TYPE